### PR TITLE
fix: update dependencies of packages in dynamic link branch

### PR DESCRIPTION
**Pull request by da1suk8**
# Description

This PR is related to https://github.com/line/cosmwasm/pull/244, https://github.com/line/cosmwasm/pull/251

Update dependencies of packages in dynamic link branch. 
(Until this commit hash [4d3ffa7db156ba4559f563c290c9a11330718f5d](https://github.com/line/wasmvm/tree/4d3ffa7db156ba4559f563c290c9a11330718f5d) has been used to do the update)
From this, dynamic link branch supports M1 Mac.

In addition to the update, the main modifications are as follows．

- added libwasmvm directory on root
- rename `GoResult` -> `GoError` 
  - https://github.com/CosmWasm/wasmvm/pull/325
- rename `into_ffi_result` -> `into_result`
- fixed `get_metrics` and `GetMetrics` test fails when the test is performed in a different environment, such as the M1 environment.  A similar error was found in `CosmWasm/wasmvm`, so the test were fixed in the same way.
  - https://github.com/CosmWasm/wasmvm/issues/340
  - https://github.com/CosmWasm/wasmvm/pull/351

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [ ] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/line/wasmvm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

**Review by zemyblue — COMMENTED** [review]
When I test in local, some errors are occurred. 
I think this error's reason is this commit, https://github.com/line/wasmvm/pull/75/commits/034d73a64240ef7e2ce4d11f3839f22ad2f11467. Please check it. 


```
=== RUN   TestGetMetrics
    lib_test.go:201: 
        	Error Trace:	lib_test.go:201
        	Error:      	Not equal: 
        	            	expected: &types.Metrics{HitsPinnedMemoryCache:0x0, HitsMemoryCache:0x0, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x56739b}
        	            	actual  : &types.Metrics{HitsPinnedMemoryCache:0x0, HitsMemoryCache:0x0, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x5001c9}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -8,3 +8,3 @@
        	            	  SizePinnedMemoryCache: (uint64) 0,
        	            	- SizeMemoryCache: (uint64) 5665691
        	            	+ SizeMemoryCache: (uint64) 5243337
        	            	 })
        	Test:       	TestGetMetrics
    lib_test.go:215: 
        	Error Trace:	lib_test.go:215
        	Error:      	Not equal: 
        	            	expected: &types.Metrics{HitsPinnedMemoryCache:0x0, HitsMemoryCache:0x1, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x56739b}
        	            	actual  : &types.Metrics{HitsPinnedMemoryCache:0x0, HitsMemoryCache:0x1, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x5001c9}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -8,3 +8,3 @@
        	            	  SizePinnedMemoryCache: (uint64) 0,
        	            	- SizeMemoryCache: (uint64) 5665691
        	            	+ SizeMemoryCache: (uint64) 5243337
        	            	 })
        	Test:       	TestGetMetrics
    lib_test.go:229: 
        	Error Trace:	lib_test.go:229
        	Error:      	Not equal: 
        	            	expected: &types.Metrics{HitsPinnedMemoryCache:0x0, HitsMemoryCache:0x2, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x1, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x56739b, SizeMemoryCache:0x56739b}
        	            	actual  : &types.Metrics{HitsPinnedMemoryCache:0x0, HitsMemoryCache:0x2, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x1, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x5001c9, SizeMemoryCache:0x5001c9}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -7,4 +7,4 @@
        	            	  ElementsMemoryCache: (uint64) 1,
        	            	- SizePinnedMemoryCache: (uint64) 5665691,
        	            	- SizeMemoryCache: (uint64) 5665691
        	            	+ SizePinnedMemoryCache: (uint64) 5243337,
        	            	+ SizeMemoryCache: (uint64) 5243337
        	            	 })
        	Test:       	TestGetMetrics
    lib_test.go:246: 
        	Error Trace:	lib_test.go:246
        	Error:      	Not equal: 
        	            	expected: &types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x2, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x1, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x56739b, SizeMemoryCache:0x56739b}
        	            	actual  : &types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x2, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x1, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x5001c9, SizeMemoryCache:0x5001c9}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -7,4 +7,4 @@
        	            	  ElementsMemoryCache: (uint64) 1,
        	            	- SizePinnedMemoryCache: (uint64) 5665691,
        	            	- SizeMemoryCache: (uint64) 5665691
        	            	+ SizePinnedMemoryCache: (uint64) 5243337,
        	            	+ SizeMemoryCache: (uint64) 5243337
        	            	 })
        	Test:       	TestGetMetrics
    lib_test.go:263: 
        	Error Trace:	lib_test.go:263
        	Error:      	Not equal: 
        	            	expected: &types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x2, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x56739b}
        	            	actual  : &types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x2, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x5001c9}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -8,3 +8,3 @@
        	            	  SizePinnedMemoryCache: (uint64) 0,
        	            	- SizeMemoryCache: (uint64) 5665691
        	            	+ SizeMemoryCache: (uint64) 5243337
        	            	 })
        	Test:       	TestGetMetrics
    lib_test.go:281: 
        	Error Trace:	lib_test.go:281
        	Error:      	Not equal: 
        	            	expected: &types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x3, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x56739b}
        	            	actual  : &types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x3, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x5001c9}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -8,3 +8,3 @@
        	            	  SizePinnedMemoryCache: (uint64) 0,
        	            	- SizeMemoryCache: (uint64) 5665691
        	            	+ SizeMemoryCache: (uint64) 5243337
        	            	 })
        	Test:       	TestGetMetrics
--- FAIL: TestGetMetrics (0.03s)


Expected :&types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x3, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x56739b}
Actual   :&types.Metrics{HitsPinnedMemoryCache:0x1, HitsMemoryCache:0x3, HitsFsCache:0x1, Misses:0x0, ElementsPinnedMemoryCache:0x0, ElementsMemoryCache:0x1, SizePinnedMemoryCache:0x0, SizeMemoryCache:0x5001c9}
```

**Comment by da1suk8** [comment]
@zemyblue 
Please tell me your environment of local PC.

```
$ rustup --version
$ sysctl machdep.cpu.brand_string
```

**Comment by zemyblue** [comment]
> @zemyblue Please tell me your environment of local PC.
> 
> ```
> $ rustup --version
> $ sysctl machdep.cpu.brand_string
> ```

```sh
$ rustup --version
rustup 1.25.1 (bb60b1e89 2022-07-12)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.64.0 (a55dd71d5 2022-09-19)`

$ sysctl machdep.cpu.brand_string
machdep.cpu.brand_string: Apple M1 Pro
```

**Comment by da1suk8** [comment]
@zemyblue 
Fixed test. Please check it.

FYI: https://github.com/CosmWasm/wasmvm/pull/351

**Review by zemyblue — APPROVED** [review]

**Review by tnasu — APPROVED** [review]

**Comment by loloicci** [comment]
[5b328d9](https://github.com/line/wasmvm/pull/75/commits/5b328d99754a1d6de1b54f37252564986eee3c88) and [bfc9e4b](https://github.com/line/wasmvm/pull/75/commits/bfc9e4b1f159e1a8e1ac96e7ec9fe4ee56eeb165) look like cherry picking from CosmWasm/wasmvm. If it is, how about writing about it in commit messages or PR descriptions?

File: api/lib_test.go [hunk 1]
```diff
@@ -197,12 +197,11 @@ func TestGetMetrics(t *testing.T) {
 
 	// GetMetrics 3
 	metrics, err = GetMetrics(cache)
-	require.NoError(t, err)
-	assert.Equal(t, &types.Metrics{
-		HitsFsCache:         1,
-		ElementsMemoryCache: 1,
-		SizeMemoryCache:     5665691,
-	}, metrics)
+	assert.NoError(t, err)
```
Comment (loloicci) [inline api/lib_test.go:200]: Why do you replace `require` with `assert`?

File: api/lib_test.go [hunk 2]
```diff
@@ -197,12 +197,11 @@ func TestGetMetrics(t *testing.T) {
...
+	require.Equal(t, uint32(0), metrics.HitsMemoryCache)
+	require.Equal(t, uint32(1), metrics.HitsFsCache)
+	require.Equal(t, uint64(1), metrics.ElementsMemoryCache)
+	require.InEpsilon(t, 5602873, metrics.SizeMemoryCache, 0.18)
```
[the remark this answers is not in the document]
Comment (da1suk8) [reply api/lib_test.go:204]: The upstream value was used.
https://github.com/CosmWasm/wasmvm/blob/9d865278deafd7d012f2347ca87eedc27586dcfa/internal/api/lib_test.go#L217

File: api/lib_test.go (hunk 1 above)
Comment (da1suk8) [reply api/lib_test.go:200]: The same fixes as upstream have been made.

https://github.com/CosmWasm/wasmvm/blob/2857d72b920e2db3ba77592e56757e7ceb7693b7/internal/api/lib_test.go#L200

**Review by shiki-tak — APPROVED** [review]

File: api/lib_test.go (hunk 1 above)
Comment (loloicci) [reply api/lib_test.go:200]: It is ok if the aim is decreasing conflicts.

File: api/lib_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (loloicci) [reply api/lib_test.go]: How about copying value from our main repository? (done with [83786a4](https://github.com/line/wasmvm/pull/75/commits/83786a4b814becab1f2d0b96a375b0a22f686f5d))

**Review by loloicci — APPROVED** [review]

**Pull request merged**